src/runtime_src/core/tools/common/Table2D.cpp

```diff
@@ -22,93 +11,120 @@
-        auto& column = m_table[i];
-        column.max_element_size = std::max(column.max_element_size, entry[i].size());
-        column.data.push_back(entry[i]);
-    }
+  // Iterate through the entry data and the table adding the table entry elements in order
+  for (size_t i = 0; i < entry.size(); ++i) {
+    auto& column = m_table[i];
+    column.max_element_size = std::max(column.max_element_size, entry[i].size());
+    column.data.push_back(entry[i]);
+  }
 }
 
 void
-Table2D::appendToOutput(std::string& output, const ColumnData& column, const std::string& data) const
+Table2D::appendToOutput(std::string& output, const std::string& prefix, const std::string& suffix, const ColumnData& column, const std::string& data) const
 {
-    // Format for table data
-    boost::format fmt("%s%s%s  ");
-    size_t left_blanks = 0;
-    size_t right_blanks = 0;
-    getBlankSizes(column, data.size(), left_blanks, right_blanks);
-    output.append(boost::str(fmt % std::string(left_blanks, ' ') % data % std::string(right_blanks, ' ')));
+  // Format for table data
+  boost::format fmt("%s%s%s%s%s");
+  size_t left_blanks = 0;
+  size_t right_blanks = 0;
+  getBlankSizes(column, data.size(), left_blanks, right_blanks);
+  output.append(boost::str(fmt % prefix % std::string(left_blanks, ' ') % data % std::string(right_blanks, ' ') % suffix));
 }
 
-std::ostream&
-Table2D::print(std::ostream& os) const
+std::string
+Table2D::toString(const std::string& prefix) const
 {
-    // Iterate through each row and column of the table to format the output
-    // Add one to account for the header row
-    for(size_t row = 0; row < m_table[0].data.size() + 1; row++) {
-        std::string output_line;
-        for (size_t col = 0; col < m_table.size(); col++) {
-            auto& column = m_table[col];
+  // Iterate through each row and column of the table to format the output
+  // Add one to account for the header row
+  std::stringstream os;
+  for (size_t row = 0; row < m_table[0].data.size() + 2; row++) {
+    std::string output_line;
+    for (size_t col = 0; col < m_table.size(); col++) {
+      auto& column = m_table[col];
 
-            // For the first row add the headers
-            if (row == 0)
-                appendToOutput(output_line, column, column.header.name);
-            // For all other output lines add the entry associated with the current row/col index
-            else
-                // Minus 1 to account for the first row being the headers
-                appendToOutput(output_line, column, column.data[row - 1]);
-        }
-        output_line.append("\n");
-        os << output_line;
+      // The first column must align with the user desires
+      // All other columns should only use the previous lines space suffix
+      const std::string column_prefix = (col == 0) ? prefix : "";
+
+      // For the first row add the headers
+      const auto space_suffix = std::string(m_inter_entry_padding, ' ');
+      if (row == 0)
```

Comment:
Being that something different needs to be done with the first and second row, to help future folks it might be better to create a switch block.  For example:
```
switch (row) {
case 0:
  appendToOutput(output_line, column_prefix, space_suffix, column, column.header.name);
  break;
case 1:
  appendToOutput(output_line, column_prefix, std::string(m_inter_entry_padding, '-'), column, std::string(column.max_element_size, '-');
  break;
default:
  appendToOutput(output_line, column_prefix, space_suffix, column, column.data[row - 2]);
  break;
}
```


default: